### PR TITLE
Add the grafana docker's src

### PR DESCRIPTION
**Pull request by YitongFeng**
I have removed the "gosu" and the docker image has passed the test at eastus cluster.

### CODE REVIEW

File: service-deployment/clusterconfig-example.yaml [hunk 1]
```diff
@@ -74,7 +74,21 @@ clusterinfo:
...
+    # port for prometheus port
+    prometheus_port: 9090
+    # The address of the grafana to connect to
+    grafana_addr: http://10.0.3.9:3000
+    # port for grafana
+    grafana_port: 3000
+    # port for node exporter
+    node_exporter_port: 9100
+    # port for node gpu-exporter
+    gexporter_port: 9202
```
Comment (fanyangCS) [inline service-deployment/clusterconfig-example.yaml]: as discussed, this PR only cover docker build & push process. please exclude this change from this PR. Same for the modification in service yaml.

File: service-deployment/src/grafana/dashboards/gpu.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+if(!_.isUndefined(ARGS.rows)) {
+  rows = parseInt(ARGS.rows, 10);
+}
+
+if(!_.isUndefined(ARGS.name)) {
+  seriesName = ARGS.name;
+}
+
+var hostName = "10.0.1.5:9100"
```
Comment (fanyangCS) [inline service-deployment/src/grafana/dashboards/gpu.js]: >10.0.1.5:9100 [](start = 16, length = 13)

we need this to be configurable. one way to do this is to make it a template

File: service-deployment/src/grafana/dockerfile.template [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+FROM debian:jessie
+MAINTAINER Yitong Feng imfengyitong@gmail.com
```
Comment (fanyangCS) [inline service-deployment/src/grafana/dockerfile.template]: remove

File: service-deployment/src/grafana/dockerfile.template [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  GRAFANA_URL={{ clusterconfig['clusterinfo']['prometheusinfo']['grafana_addr'] }} \
+  PROM_URL={{ clusterconfig['clusterinfo']['prometheusinfo']['prometheus_addr'] }} \
+  GRAFANA_VERSION=4.6.2 \
+  GF_PLUGIN_DIR=/grafana-plugins \
+  GF_PATHS_LOGS=/var/log/grafana \
+  GF_PATHS_DATA=/var/lib/grafana \
+  UPGRADEALL=false
+
+
+COPY ./run.sh /run.sh
```
Comment (fanyangCS) [inline service-deployment/src/grafana/dockerfile.template]: >/ [](start = 14, length = 1)

root isn't a good workdir. I suggest you to set a workdir first and copy files to workdir

File: service-deployment/src/grafana/dockerfile.template [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+COPY ./grafana_config.sh /grafana_config.sh
+COPY ./start_server.sh /start_server.sh
+COPY ./prom-datasource.json /
+COPY ./dashboards/*.json /dashboards/
+
+RUN \
+  apt-get update && \
+  apt-get -y --force-yes --no-install-recommends install libfontconfig curl ca-certificates git jq && \
+  curl https://s3-us-west-2.amazonaws.com/grafana-releases/release/grafana_${GRAFANA_VERSION}_amd64.deb > /tmp/grafana.deb && \
```
Comment (fanyangCS) [inline service-deployment/src/grafana/dockerfile.template]: >s3-us-west-2.amazonaws.com [](start = 15, length = 26)

I doubt this the official place to get grafana. if cannot get it through apt-get, I suggest to find it from grafana official site

File: service-deployment/src/grafana/dockerfile.template [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  dpkg -i /tmp/grafana.deb && \
+  rm -f /tmp/grafana.deb && \
+  curl -L https://github.com/tianon/gosu/releases/download/1.10/gosu-amd64 > /usr/sbin/gosu && \
+  chmod +x /usr/sbin/gosu && \
```
Comment (fanyangCS) [inline service-deployment/src/grafana/dockerfile.template]: we can't do this.

File: service-deployment/src/grafana/dashboards/gpu.js (hunk 2 above)
Comment (YitongFeng) [reply service-deployment/src/grafana/dashboards/gpu.js]: Solved. It is an initial configuration, set it to empty is ok.

File: service-deployment/src/grafana/dockerfile.template (hunk 4 above)
Comment (YitongFeng) [reply service-deployment/src/grafana/dockerfile.template]: Solved

File: service-deployment/src/grafana/dockerfile.template (hunk 5 above)
Comment (YitongFeng) [reply service-deployment/src/grafana/dockerfile.template]: It's official site.

File: service-deployment/clusterconfig-example.yaml (hunk 1 above)
Comment (YitongFeng) [reply service-deployment/clusterconfig-example.yaml]: Keep the same with master branch now.

File: service-deployment/src/grafana/dockerfile.template (hunk 3 above)
Comment (YitongFeng) [reply service-deployment/src/grafana/dockerfile.template]: Solved.

File: service-deployment/src/grafana/dockerfile.template [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+FROM debian:jessie
+
+### GRAFANA_VERSION=latest = nightly build
+ENV \
+  GRAFANA_URL={{ clusterconfig['clusterinfo']['prometheusinfo']['grafana_addr'] }} \
```
Comment (ydye) [inline service-deployment/src/grafana/dockerfile.template]: hardcode. Don't hadrcode IP address in the docker image

File: service-deployment/src/grafana/dockerfile.template [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  PROM_URL={{ clusterconfig['clusterinfo']['prometheusinfo']['prometheus_addr'] }} \
```
Comment (ydye) [inline service-deployment/src/grafana/dockerfile.template]: hardcode. Don't hadrcode IP address in the docker image

File: service-deployment/src/grafana/dockerfile.template (hunk 7 above)
Comment (YitongFeng) [reply service-deployment/src/grafana/dockerfile.template]: Solved

File: service-deployment/src/grafana/dockerfile.template (hunk 8 above)
Comment (YitongFeng) [reply service-deployment/src/grafana/dockerfile.template]: Solved

**Pull request closed without merging**